src/aiida/schedulers/plugins/direct.py

```diff
@@ -354,9 +354,18 @@ def _parse_submit_output(self, retval, stdout, stderr):
 
         return stdout.strip()
 
-    def _get_kill_command(self, jobid):
-        """Return the command to kill the job with specified jobid."""
-        submit_command = f'kill {jobid}'
+    def _get_kill_command(self, process_id):
+        """Return the command to kill the process with specified id and all its descendants."""
+        from psutil import Process
+
+        # get a list of the process id of all descendants
+        process = Process(int(process_id))
+        children = process.children(recursive=True)
+        process_ids = [process_id]
+        process_ids.extend([str(child.pid) for child in children])
+        process_ids_str = ' '.join(process_ids)
+
+        submit_command = f'kill {process_ids_str}'
```

Comment:
Just a side node:
I've encountered cases where `kill PID` silently returns without actually killing a job. 
I would suggest handling this scenario, if PID still exists after sending the command `kill PID`. 
then properly inform with a log message.

Reply:
In this case kill should return a nonzero exit code which should then be caught here https://github.com/aiidateam/aiida-core/blob/f57591655a8f91d1b37bd0f46e8c6dc6a9566c2c/src/aiida/schedulers/plugins/direct.py#L365
As you can see can in the usage of this function in the bash scheduler (that is the base class of the direct one) https://github.com/aiidateam/aiida-core/blob/f57591655a8f91d1b37bd0f46e8c6dc6a9566c2c/src/aiida/schedulers/plugins/bash.py#L73-L74